site/src/pages/UserExternalAuthSettingsPage/UserExternalAuthSettingsPage.tsx

```diff
@@ -0,0 +1,97 @@
+import { FC, useState } from "react";
+import { UserExternalAuthSettingsPageView } from "./UserExternalAuthSettingsPageView";
+import {
+  listUserExternalAuths,
+  unlinkExternalAuths,
+  validateExternalAuth,
+} from "api/queries/externalauth";
+import { Section } from "components/SettingsLayout/Section";
+import { DeleteDialog } from "components/Dialogs/DeleteDialog/DeleteDialog";
+import { useMutation, useQuery, useQueryClient } from "react-query";
+import { displayError, displaySuccess } from "components/GlobalSnackbar/utils";
+import { getErrorMessage } from "api/errors";
+
+const UserExternalAuthSettingsPage: FC = () => {
+  const queryClient = useQueryClient();
+  // This is used to tell the child components something was unlinked and things
+  // need to be refetched
+  const [unlinked, setUnlinked] = useState(0);
+
+  const {
+    data: externalAuths,
+    error,
+    isLoading,
+    refetch,
+  } = useQuery(listUserExternalAuths());
+
+  const [appToUnlink, setAppToUnlink] = useState<string>();
+  const mutateParams = unlinkExternalAuths(queryClient);
+  const unlinkAppMutation = useMutation({
+    ...mutateParams,
+    onSuccess: async () => {
+      await mutateParams.onSuccess();
+      await refetch();
+      setUnlinked(unlinked + 1);
+    },
+  });
+
+  const validateAppMutation = useMutation(validateExternalAuth(queryClient));
+
+  return (
+    <Section title="External Authentication">
+      <UserExternalAuthSettingsPageView
+        isLoading={isLoading}
+        getAuthsError={error}
+        auths={externalAuths}
+        unlinked={unlinked}
+        onUnlinkExternalAuth={(providerID: string) => {
+          setAppToUnlink(providerID);
+        }}
+        onValidateExternalAuth={async (providerID: string) => {
+          try {
+            await validateAppMutation.mutateAsync(providerID, {
```

Comment:
Since you are using `mutateAsync` you don't need to use the `onSuccess`. 

You can just do: 
```ts
try {
  const data = await validateAppMutation.mutateAsync(providerID)
  if (data.authenticated) {
    displaySuccess("Application link is valid.");
  } else {
    displayError(
      "Application link is not valid. Please unlink the application and reauthenticate.",
    );
  }
} catch {
  ...
}
```